modules/tinymce/src/themes/silver/main/ts/ui/menus/item/build/ColorSwatchItem.ts

```diff
@@ -34,6 +42,9 @@ export const renderColorSwatchItem = (spec: Menu.ColorSwatchMenuItem, backstage:
     movement: deriveMenuMovement(columns, presets)
   };
 
+  // TINY-10806: Avoid duplication of ARIA role="menu" in the accessibility tree for Color Swatch menu item.
+  const widgetSketchSpec = removeRoleAttr(AlloyMenu.sketch(widgetSpec));
```

Comment:
Why are we post-mutating the spec, and not just avoiding the role going into the spec to begin with?

Reply:
You probably need to modify alloy to allow a menu to not have the menu role. This color swatch menu item is very special it kind of works like a menu since it has child elements with the role=menuitemradio but since the parent is already a role=menu it shouldn't have role=menu on it's container. Maybe add some weird boolean `menuRole` that defaults to true then you can set it to false and if you do then you. don't add the role=menu in the MenuSpec.ts. That way it's all immutable and passed down to the thing.

Reply:
Initially, I'd like to avoid extending the menu spec for such an exceptional case. But now I see it is the wrong approach, so I added the `menuRole` boolean field defaulted to `true`.

Reply:
If it was a public API that might've been worth considering, but when it's all internal like this then yes adjusting the API is fine - particularly if it avoids mutating what is basically a config object.

Side effects, implicit dependencies, spaghetti code, these are all problems that come up in a few years if architecture decisions aren't taken carefully :)